File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.java
class Solution {
    public int smallestNumber(int n, int t) {
        int product=t+1, n_copy=n,digit=0;
        while(product %t !=0){
            product=1;
            n=n_copy;
            n_copy++;
            while(n>0){
                digit=n%10;
                product *= digit;
                n /=10;
            }
            n=n_copy-1;
        }
        
        return n;
    }
}

File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.c
int smallestNumber(int n, int t) {
    int digit=0,dp=1,n_og=n;
    while(dp%t!=0){
        n=n_og;
        dp=1;
        n_og++;
        while(n>0){
            digit=n%10;
            dp *= digit;
            n /=10;
        }
        n=n_og-1;
    }
    return n;
}